src/parser/bgp/attributes/mod.rs

```diff
@@ -189,30 +179,28 @@ impl AttributeParser {
                 AttrType::DEVELOPMENT => {
                     let mut value = vec![];
                     for _i in 0..attr_length {
-                        value.push(attr_data.get_u8());
+                        value.push(attr_data.read_u8()?);
                     }
                     Ok(AttributeValue::Development(value))
                 }
                 AttrType::ONLY_TO_CUSTOMER => parse_only_to_customer(attr_data),
-                _ => Err(ParserError::Unsupported(format!(
-                    "unsupported attribute type: {:?}",
-                    attr_type
-                ))),
+                // TODO: Should it be treated as a raw attribute instead?
+                _ => Err(ParserError::UnsupportedAttributeType(attr_type)),
             };
 
             match attr {
                 Ok(value) => {
                     assert_eq!(attr_type, value.attr_type());
                     attributes.push(Attribute { value, flag });
                 }
+                Err(e) if partial => {
+                    // TODO: Is this correct? If we don't have enough bytes, split_to would panic.
+                    // it's ok to have errors when reading partial bytes
+                    warn!("PARTIAL: {}", e);
```

Comment:
This should be fine. However, if I recall correctly, the previous version would not panic because it maintains the number of bytes it holds available to read. we should probably double-check to make sure we don't panic in this case.

Reply:
You are right, it would not panic. However, it does propagate the error that there are not enough bytes remaining before it  reaches this code.

Reply:
I don't have a good idea on this issue. My suggestion is to keep this note here and come back to this if we see panics later on with the `PARTIAL` warning.